### ISSUE TITLE
Upstream image not found

### ISSUE THREAD

**Issue by alledm** [cloud66/habitus#37]
Hello,

I have a set of images that depend on an image "go_base".
I am affected by something similar to #34 reported by @kamerblauwlicht 

I am running 0.4.7 and I still see this happening.

```
2016/10/19 12:58:53 Parallel build for go_base
2016/10/19 12:58:53 Building go_base
2016/10/19 12:58:53 Parsing and converting 'Dockerfile.go_base'
2016/10/19 12:58:53 Building the go_base-66b52139 image from Dockerfile.go_base.generated

[cut]

2016/10/19 12:59:07  Writing the new Dockerfile into Dockerfile.message_store.generated 
2016/10/19 12:59:07 Building the go_message_store-66b52139 image from Dockerfile.message_store.generated
Step 1 : FROM go_base
Step 1 : FROM go_base
Pulling repository docker.io/library/go_base
Pulling repository docker.io/library/go_base
2016/10/19 12:59:12  Build for step go_message_store failed due to Error: image library/go_base:latest not found
Build step 'Execute shell' marked build as failure
```

this is intermittent and re-running the job usually fixes it.

**Comment by alledm** [comment]
This is actually now affecting me every single time, re-running it does not work anymore..
Will try adding a sleep command after the build.

**Comment by alledm** [comment]
Adding a sleep command didn't work.
I will have to stop using habitus and go back to the old workflow until this issue has been fixed.

Currently all my builds are failing.

I have  a relatively simple setup but I cannot get it to work without failing.

```
build:
  version: 2016-03-14 # version of the build schema.
  steps:
    # If you modify steps make sure you also edit jenkins/habitus_build.sh
    builder:
      name: go_builder
      dockerfile: Dockerfile.go_builder
    godoc:
      name: godoc
      dockerfile: Dockerfile.godoc
    health:
      name: health_base
      dockerfile: Dockerfile.health_base
      command: sleep 2
    frontend_api:
      name: frontend_api
      dockerfile: Dockerfile.frontend_api
      depends_on:
        - health
    gl7:
      name: gl7
      dockerfile: Dockerfile.gl7
      depends_on:
        - health
    ari:
      name: ari
      dockerfile: Dockerfile.ari
      depends_on:
        - health
    ale:
      name: ale
      dockerfile: Dockerfile.ale
      depends_on:
        - health
    base:
      name: go_base
      dockerfile: Dockerfile.go_base
      depends_on:
        - builder
      command: sleep 2
    message_store:
      name: go_message_store
      dockerfile: Dockerfile.message_store
      depends_on:
        - base
    ghir_grpc:
      name: go_ghir_grpc
      dockerfile: Dockerfile.ghir_grpc
      depends_on:
        - base
    persistence:
      name: go_persistence
      dockerfile: Dockerfile.persistence
      depends_on:
        - base
```

**Comment by kamerblauwlicht** [comment]
hi @alledm. looks like the go docker client we are using can't handle parallel builds the correct way.   let me investigate.

**Comment by alledm** [comment]
thank you!

**Comment by alledm** [comment]
Hello @kamerblauwlicht, where you able to make progress on this issue? Is there anything I can do to help?

**Comment by alledm** [comment]
Hi,

thank for your support but I am switching to use plain jenkins pipelines for this.
regards.

**Comment by kamerblauwlicht** [comment]
sorry to hear. we are improving habitus. I keep this ticket open to improve habitus.

**Comment by kamerblauwlicht** [comment]
fixed by #62
